src/main/c/Objects/pyjobject.c

```diff
@@ -77,8 +78,8 @@ static void pyjobject_dealloc(PyJObject *self)
             (*env)->DeleteGlobalRef(env, self->clazz);
         }
     }
-
-    Py_TYPE((PyObject*) self)->tp_free((PyObject*) self);
+    tp->tp_free((PyObject*) self);
```

Comment:
Why do we call tp_free() on this one?

Reply:
PyJObject and the subtypes that we create manually like PyJArray and PyJClass do not use garbage collection because they can't be involved in reference cycles so they need to be freed with `PyObject_Del`. When we dynamically make subtypes for subclasses, like java.util.ArrayList or any other Java class then python automatically adds cgarbage collection so `PyObject_GC_Del` needs to be called to free the memory. `tp_free` is always set to the correct option from those two so we don't have to try anything else to figure out which one is right.